src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala

```diff
@@ -674,4 +675,32 @@ final case class EServicesApiServiceImpl(
       }
     }
   }
+
+  override def deleteEService(
+    eServiceId: String
+  )(implicit contexts: Seq[(String, String)], toEntityMarshallerProblem: ToEntityMarshaller[Problem]): Route = {
+
+    def deleteEserviceIfEmpty(eService: CatalogProcessEService): Future[Boolean] =
+      if (!eService.descriptors.exists(_.state != DRAFT))
+        catalogProcessService.deleteEService(eService.id).map(_ => true)
+      else Future.successful(false)
```

Comment:
Hadn't we changed this logic together last week?

Reply:
If I remember right, we reviewed together this other piece of code: [getProducerEServices](https://github.com/pagopa/interop-be-backend-for-frontend/blob/df9b31f69ba69dd43c031bee42db3e18f3368677/src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala#L278).

We wrote this one with the help of @galales!